### PR TITLE
adding init fn tests

### PR DESCRIPTION
**Pull request by vchiley**
adds tests for init functions
param init throws error if params are left uninitialized
https://github.com/mosaicml/examples/pull/204#issuecomment-1454891619
https://mosaicml.atlassian.net/browse/RESEARCH-471

### CODE REVIEW

File: examples/llm/tests/test_init_fn.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                        init_fn_=nn.init.ones_))
+
+    # verify layer norm is init to bias=0 and weight=1
+    assert (model.ln_1.weight == 1).all()
+    if model.ln_1.bias is not None:
+        assert (model.ln_1.bias == 0).all()
+
+    # verify _is_residual works
+    expected_value = 1 / math.sqrt(2 * cfg.n_layers) if is_residual else 1
+    print(expected_value)
```
Comment (dakinggg) [inline examples/llm/tests/test_init_fn.py:56]:
```suggestion
```

**Review by dakinggg — APPROVED** [review]

File: examples/llm/tests/test_init_fn.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+import pytest
+import torch
+from composer.utils import reproducibility
+from omegaconf import OmegaConf as om
+from torch import nn
+
+from examples.llm.src.models.param_init_fns import generic_param_init_fn_
+
+
+class EgMLP(nn.Module):
```
[the remark this answers is not in the document]
Comment (vchiley) [reply examples/llm/tests/test_init_fn.py:16]: abbreviation for exempli gratia: a Latin phrase that means "for example."

but as I write that, I admit, its probably not the best name for it; I'll change it to just `MLP`

File: examples/llm/src/models/param_init_fns.py [hunk 3]
```diff
@@ -105,6 +113,18 @@ def generic_param_init_fn_(module, cfg, init_fn_):
         if module.out_proj.bias is not None:
             torch.nn.init.zeros_(module.out_proj.bias)
 
+    else:
+        error_str = []
+        for n, p in module.named_parameters(recurse=False):
```
[the remark this answers is not in the document]
Comment (vchiley) [reply examples/llm/src/models/param_init_fns.py:118]: Ok, I'll remove it.

File: examples/llm/tests/test_init_fn.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dakinggg) [reply examples/llm/tests/test_init_fn.py]: haha, oh _that_ eg :D

**Pull request merged**